app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -285,14 +293,22 @@ async function addEthereumChainHandler(
         networkConfigurationId,
       },
     });
-    await setActiveNetwork(networkConfigurationId);
   } catch (error) {
     // For the purposes of this method, it does not matter if the user
     // declines to switch the selected network. However, other errors indicate
     // that something is wrong.
     if (error.code !== errorCodes.provider.userRejectedRequest) {
       return end(error);
     }
+  } finally {
+    endApprovalFlow({ id: approvalFlowId });
+  }
+
+  try {
+    await setActiveNetwork(networkConfigurationId);
```

Comment:
I moved this outside of the approval flow, as otherwise the "Switched to network X" notification appears on top of the approval flow loading page for a long amount of time.